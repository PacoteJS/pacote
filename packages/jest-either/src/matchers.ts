export interface AsymmetricMatcher {
  asymmetricMatch(other: any): boolean
}

export function isAsymmetricMatcher(
  matcher: any
): matcher is AsymmetricMatcher {
  return typeof matcher.asymmetricMatch === 'function'
}
