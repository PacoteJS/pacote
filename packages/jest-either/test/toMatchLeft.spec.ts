import { left, right } from 'fp-ts/lib/Either'
import matchers from '../src/index'

expect.extend(matchers)

describe('.toMatchLeft()', () => {
  test('passes when spec matches the object on the left', () => {
    expect(left({ a: 1, b: 2 })).toMatchLeft({ a: 1 })
  })

  test('passes when spec deeply matches the object on the left', () => {
    expect(left({ a: 1, b: { c: [2], d: 3 } })).toMatchLeft({ b: { c: [2] } })
  })

  test('passes when regular expression matches a string on the left', () => {
    expect(left('aa')).toMatchLeft(/a/)
  })

  test('fails when spec does not match the object on the left', () => {
    expect(() => expect(left({ a: 1, b: 2 })).toMatchLeft({ a: 2 })).toThrow()
  })

  test('fails when object is a right', () => {
    expect(() => expect(right({ a: 1 })).toMatchLeft({ a: 1 })).toThrow()
  })

  test('fails when regular expression does not match a string on the left', () => {
    expect(() => expect(left('aa')).toMatchLeft(/b/)).toThrow()
  })
})

describe('.not.toMatchLeft()', () => {
  test('passes when object is a right', () => {
    expect(right('right')).not.toMatchLeft('left')
  })

  test('passes when spec does not match the object on the left', () => {
    expect(left({ a: 1, b: 2 })).not.toMatchLeft({ a: 2 })
  })

  test('passes when regular expression does not match a string on the left', () => {
    expect(left('aa')).not.toMatchLeft(/b/)
  })

  test('fails when spec deeply matches the object on the left', () => {
    expect(() =>
      expect(left({ a: 1, b: { c: 2, d: 3 } })).not.toMatchLeft({ b: { c: 2 } })
    ).toThrow()
  })

  test('fails when spec matches the object on the left', () => {
    expect(() => expect(left({ a: 1 })).not.toMatchLeft({ a: 1 })).toThrow()
  })

  test('fails when regular expression matches a string on the left', () => {
    expect(() => expect(left('aa')).not.toMatchLeft(/a/)).toThrow()
  })
})
