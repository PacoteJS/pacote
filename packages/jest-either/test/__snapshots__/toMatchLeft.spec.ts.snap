// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toMatchLeft() fails when object deeply matches the left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toMatchLeft([22m[32mexpectedLeft[39m[2m)[22m

Expected Either not to match left:
  [32m{\\"b\\": {\\"c\\": 2}}[39m

Received left:
  [31m{\\"a\\": 1, \\"b\\": {\\"c\\": 2, \\"d\\": 3}}[39m"
`;

exports[`.not.toMatchLeft() fails when object matches the left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toMatchLeft([22m[32mexpectedLeft[39m[2m)[22m

Expected Either not to match left:
  [32m{\\"a\\": 1}[39m

Received left:
  [31m{\\"a\\": 1}[39m"
`;

exports[`.toMatchLeft() fails when object does not match the left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchLeft([22m[32mexpectedLeft[39m[2m)[22m

Expected Either to match left:
  [32m{\\"a\\": 2}[39m

Received left:
  [31m{\\"a\\": 1, \\"b\\": 2}[39m"
`;

exports[`.toMatchLeft() fails when object is a right 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchLeft([22m[32mexpectedLeft[39m[2m)[22m

Expected Either to match left:
  [32m{\\"a\\": 1}[39m

Received right."
`;
